src/lib.rs

```diff
@@ -136,7 +136,7 @@ mod rsa;
 
 pub mod signature;
 
-#[cfg(any(feature = "use_heap", test))]
+#[cfg(feature = "use_heap")]
```

Comment:
Consider the case where the library will be linked into an executable that doesn't have `malloc`. That is what disabling `use_heap` is supposed to allow.

But, we still want people to be able to run the *ring* test suite for that configuration, in another executable (the one generated by `cargo test`. Right now, that means they have to have a working `malloc`. In the future, we'll hopefully remove this requirement.

tl;dr: I think the original condition is what we want.